File: MS.NET/Fundamentals/Language/OOP/interfacetest.cs
using Banking;
using System;

static class Program
{
	private static void PayQuarterlyInterest(Account[] accounts)
	{
		foreach(Account acc in accounts)
		{
			IProfitable p = acc as IProfitable;
			if(p != null)
			{
				double interest = p.GetInterest(3);
				acc.Deposit(interest);
			}
		}
	}

	private static void DeductAnnualTax(Account[] accounts)
	{
		foreach(Account acc in accounts)
		{
			ITaxable t = acc as ITaxable;
			t?.Deduct(); //if(t != null) t.Deduct()
		}
	}

	public static void Main()
	{
		Account[] bank = new Account[5];
		bank[0] = Banker.OpenSavingsAccount();
		bank[0].Deposit(5000);
		bank[1] = Banker.OpenCurrentAccount();
		bank[1].Deposit(20000);
		bank[2] = Banker.OpenSavingsAccount();
		bank[2].Deposit(25000);
		bank[3] = Banker.OpenCurrentAccount();
		bank[3].Deposit(40000);
		bank[4] = Banker.OpenSavingsAccount();
		bank[4].Deposit(45000);

		PayQuarterlyInterest(bank);
		DeductAnnualTax(bank);

		foreach(var acc in bank)
			Console.WriteLine($"{acc.Id}\t{acc.Balance}");


	}
}

File: Java/Fundamentals/Language/OOP/InterfaceTest.java
import edu.met.bkc.banking.*;

class InterfaceTest{

	private static void payQuarterlyInterest(Account[] accounts){
		for(Account acc : accounts){
			if(acc instanceof Profitable){
				Profitable p = (Profitable)acc;
				double interest = p.getQuarterlyInterest();
				acc.deposit(interest);
			}
		}
	}

	public static void main(String[] args){
		Account[] bank = new Account[4];
		bank[0] = Banker.openSavingsAccount();
		bank[0].deposit(5000);
		bank[1] = Banker.openCurrentAccount();
		bank[1].deposit(20000);
		bank[2] = Banker.openCurrentAccount();
		bank[2].deposit(30000);
		bank[3] = Banker.openSavingsAccount();
		bank[3].deposit(35000);
		payQuarterlyInterest(bank);
		for(Account acc : bank)
			System.out.printf("%d\t%.2f%n", acc.getId(), acc.getBalance());
	}
}

